### PR TITLE
graphicsmagick under the hood except for psd files and imagemagick's convert -flatten

### PR DESCRIPTION
**Pull request by jenlindner**

### CODE REVIEW

**Comment by scossu** [comment]
What is the issue with gm and `convert -flatten`?

**Comment by jenlindner** [comment]
It's because gm uses mogrify to do those operations by default, but doesn't
support the -flatten argument. Another way around it is to use gm's convert
for the same thing:

https://github.com/minimagick/minimagick/issues/392

It seems worthwhile and I started working on a test for that today too.

Jen

File: app/models/file_set.rb [hunk 1]
```diff
@@ -17,9 +17,16 @@ def create_derivatives(filename, opts = {})
     # pdf for text assets
     # second thumbnail for CITI
     def create_asset_derivatives(filename)
+      # byebug
```
Comment (awead) [inline app/models/file_set.rb]: remove.

File: app/models/file_set.rb [hunk 2]
```diff
@@ -17,9 +17,16 @@ def create_derivatives(filename, opts = {})
...
       case mime_type
       when *self.class.image_mime_types
-        Hydra::Derivatives::ImageDerivatives.create(filename, outputs: image_outputs)
+        if mime_type =~ /psd/i || mime_type =~ /vnd.adobe.photoshop/i
+          MiniMagick.with_cli(:imagemagick) do
+            Hydra::Derivatives::ImageDerivatives.create(filename, outputs: image_outputs)
+          end
+        else
+          Hydra::Derivatives::ImageDerivatives.create(filename, outputs: image_outputs)
+        end
```
Comment (awead) [inline app/models/file_set.rb:24]: why not:
``` ruby
when *image_mime_types_with_cli
  MiniMagick.with_cli(:imagemagick) do
    Hydra::Derivatives::ImageDerivatives.create(filename, outputs: image_outputs)
  end
when *self.class.image_mime_types
  Hydra::Derivatives::ImageDerivatives.create(filename, outputs: image_outputs)
[...]

def image_mime_types_with_cli
  ["psd", "vnd.adobe.photoshop"]
end
```
This makes the method less branchy. Alternatively, you could define `image_mime_types_with_cli` in config/initializers/hydra_works.rb if you need to use these mime types outside of this class.

File: spec/models/file_set/file_set_derivatives_spec.rb [hunk 3]
```diff
@@ -46,6 +47,19 @@
         ]
       end
 
+      context "with a psd mime type" do
+        before {
```
Comment (awead) [inline spec/models/file_set/file_set_derivatives_spec.rb]: use do/end with multilines. {} are reserved for one-liners.

**Review by awead — CHANGES_REQUESTED** [review]

File: app/models/file_set.rb (hunk 2 above)
Comment (jenlindner) [reply app/models/file_set.rb:24]: Yes, cleaner.

File: spec/models/file_set/file_set_derivatives_spec.rb (hunk 3 above)
Comment (jenlindner) [reply spec/models/file_set/file_set_derivatives_spec.rb]: Right.

**Review by awead — APPROVED** [review]

**Pull request merged**